test/e2e/providers/exec/docker.go

```
@@ -42,10 +45,14 @@ func RunDockerExecCommand(container, cmd, workdir string, envs map[string]string
 	}
 	args = append(args, "-w", workdir, container)
 
-	if strings.Contains(cmd, "/bin/sh") {
-		// Just split in to "/bin/sh" "-c" and "actual_cmd"
+	if strings.Contains(cmd, "/bin/sh") || strings.Contains(cmd, "bash") {
+		// Split in to tokens using shell-style rules for quoting and commenting.
 		// This is useful for passing piped commands in to os/exec interface.
-		args = append(args, strings.SplitN(cmd, " ", 3)...)
+		tokens, err := shlex.Split(cmd)
+		if err != nil {
+			log.Fatal(err)
```

why not return the error